### ISSUE TITLE
After updating v0.108.0-b.19 errors in DNS Config

### ISSUE THREAD

**Issue by whyisthisbroken** [AdguardTeam/AdGuardHome#5052]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

DietPi

### CPU architecture

ARMv7

### Installation

GitHub releases or script from README

### Setup

On one machine RPi 4 8Gb

### AdGuard Home version

v0.108.0-b.19

#### What did you do?
I updatet the v0.108.0-b.18 to v0.108.0-b.19
#### Expected result
working DNS Settings
#### Actual result
not working DNS Settings

All other Settings work, but if i save any settings in the DNS Settings like Cache / Server configuration/ Upstream etc, the DNS Server kills itself, no more internet and thats it.

I go back to b18 from the AGH Backup - i dont know whats happened here....

![Screenshot 2022-10-19 183618](https://user-images.githubusercontent.com/96284773/196751931-f4a2e67b-7d05-4abc-ac3e-62658c654ed8.jpg)

**Comment by EugeneOne1** [comment]
@whyisthisbroken, hello and thanks for the report.  Could you please check out the latest edge build?  It shouldn't block reconfiguring the server now.

Also, it'd be really helpful if you'll share your `AgGuardHome.yaml` configuration file (redacted if needed) to us via devteam@adguard.com.  We're particularly interested in the DNS configuration causing these errors.

**Comment by whyisthisbroken** [comment]
I will do this later if I'm back from work. 👍🏻

**Comment by whyisthisbroken** [comment]
So - i´ve installed the latest edge version and the errors are gone. Restartet my RPi and everything works so far.
Here is my YAML File from the Beta 19, i´ve compared it to the new edge version and there are no changes... (also not to b18)
maybe it´s not a yaml problem.

[v0.108.0-b.19 AdGuardHome.zip](https://github.com/AdguardTeam/AdGuardHome/files/9839488/v0.108.0-b.19.AdGuardHome.zip)

**Comment by EugeneOne1** [comment]
@whyisthisbroken, it's definitely not, but some bugs are only reproducable within a specific configuration.  So it's more about reproducing the setup as close as possible.  I'll close the issue for now, but feel free to reopen it if the problem occured again.  Perhaps, we'll do it ourselves after the investigation.  Thanks for your help.

**Comment by druid-heifeng** [comment]
docker x86 0.108.0-b.19

same issue, normal after switching to edge version.

here is my yaml 
[adg.yaml.zip](https://github.com/AdguardTeam/AdGuardHome/files/9916650/adg.yaml.zip)

**Comment by druid-heifeng** [comment]
here is log

```
2022/11/02 02:37:05.986529 [info] Entering the UDP listener loop on [::]:53
2022/11/02 02:37:05.986580 [info] Entering the tcp listener loop on [::]:53
2022/11/02 02:37:47.681006 [info] saving filter 1667356617 contents to: /opt/adguardhome/work/data/filters/1667356617.txt
2022/11/02 02:37:47.681108 [info] updated filter 1667356617: 1123989 bytes, 57135 rules
2022/11/02 02:37:56.969556 [error] deleting filter "/opt/adguardhome/work/data/filters/2.txt": rename /opt/adguardhome/work/data/filters/2.txt /opt/adguardhome/work/data/filters/2.txt.old: no such file or directory
2022/11/02 02:38:04.706331 [info] saving filter 1667356618 contents to: /opt/adguardhome/work/data/filters/1667356618.txt
2022/11/02 02:38:04.706424 [info] updated filter 1667356618: 274726 bytes, 7357 rules
2022/11/02 05:28:54.992686 [info] Start reconfiguring the server
2022/11/02 05:28:54.992707 [info] Stopping the DNS proxy server
2022/11/02 05:28:54.993678 [info] Stopped the DNS proxy server
2022/11/02 05:28:54.993923 [error] POST 192.168.88.16:5380 /control/dns_config: could not reconfigure the server: stopping DNS server: 2 errors: "could not stop primary resolvers properly: stopping dns proxy server: failed to close some upstreams: 2 errors: \"failed to close some connections: tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:60106->101.101.101.101:853: write: broken pipe\", \"failed to close some connections: 7 errors: \\\"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:47058->1.1.1.1:853: write: broken pipe\\\", \\\"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:52774->1.1.1.1:853: write: broken pipe\\\", \\\"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:52808->1.1.1.1:853: write: broken pipe\\\", \\\"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:52810->1.1.1.1:853: write: broken pipe\\\", \\\"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:53176->1.1.1.1:853: write: broken pipe\\\", \\\"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:53534->1.1.1.1:853: write: broken pipe\\\", \\\"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 172.17.0.6:53524->1.1.1.1:853: write: broken pipe\\\"\"", "could not stop internal resolvers properly: failed to close some upstreams: 2 errors: \"failed to close some connections: 10 errors: \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\"\", \"failed to close some connections: 14 errors: \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\", \\\"use of closed network connection\\\"\""
```
